### ISSUE TITLE
Tensorflow Checkpoint and CKPT is not working and does not give any readable error.

### ISSUE THREAD

**Issue by StructML** [tensorflow/tensorflow#3821]
### Environment info
Operating System:

Mac OSX El Capitan

If installed from binary pip package, provide:

1. Which pip package you installed.

The latest version of Tensorflow

2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.

0.10.0rc0

### Steps to reproduce

I have trained a convolutional model through the main program and stored it into a checkpoint and ckpt file. The problem lies in the evaluation program. The ckpt file seems to not output anything but a bunch of errors. The program does not complete either.

The code for main is:

    import Input
    import Process

    import time
    import numpy as np

    import tensorflow as tf
    from datetime import datetime

    FLAGS = tf.app.flags.FLAGS

    def train():
        with tf.Session() as sess:
            images, labels = Process.inputs()

            forward_propgation_results = Process.forward_propagation(images)

            train_loss, cost = Process.error(forward_propgation_results, labels)

            image_summary_t = tf.image_summary(images.name, images, max_images = 2)

            summary_op = tf.merge_all_summaries()

            init = tf.initialize_all_variables()

            saver = tf.train.Saver()

            sess.run(init)

            saver = tf.train.Saver(tf.all_variables())

            tf.train.start_queue_runners(sess = sess)

            train_dir = "/Users/Zanhuang/Desktop/NNP/model.ckpt"

            summary_writer = tf.train.SummaryWriter(train_dir, sess.graph)

            for step in range(100):
                start_time = time.time()
                print(sess.run([train_loss, cost]))
                duration = time.time() - start_time
                if step % 1 == 0:
                    num_examples_per_step = FLAGS.batch_size
                    examples_per_sec = num_examples_per_step / duration
                    sec_per_batch = float(duration)

                    format_str = ('%s: step %d, (%.1f examples/sec; %.3f ''sec/batch)')
                    print (format_str % (datetime.now(), step, examples_per_sec, sec_per_batch))

                    summary_str = sess.run(summary_op)
                    summary_writer.add_summary(summary_str, step)

                    if step % 2 == 0:
                        checkpoint_path = train_dir
                        saver.save(sess, checkpoint_path, global_step = step)

    def main(argv = None):
        train()

    if __name__ == '__main__':
      tf.app.run()

and the eval is 

    import tensorflow as tf

    import main
    import Process
    import Input
    
    eval_dir = "/Users/Zanhuang/Desktop/NNP/model.ckpt-98"
    checkpoint_dir = "/Users/Zanhuang/Desktop/NNP/checkpoint"

    def evaluate():
      with tf.Graph().as_default() as g:
        images, labels = Process.eval_inputs()
        forward_propgation_results = Process.forward_propagation(images)
        init_op = tf.initialize_all_variables()
        saver = tf.train.Saver()
        top_k_op = tf.nn.in_top_k(forward_propgation_results, labels, 1)

      with tf.Session(graph = g) as sess:
        tf.train.start_queue_runners(sess = sess)
        sess.run(init_op)
        saver.restore(sess, eval_dir)
        for i in range(100):
            print(sess.run(top_k_op))

    def main(argv = None):
        evaluate()

    if __name__ == '__main__':
      tf.app.run()

### What have you tried?
1. I tried initializing the variables before running the queue's but that only removed the errors. The errors also do not point to any problem thats in the code.

### Logs or other output that would be helpful
(If logs are large, please upload as attachment).

The output is:

[error.txt](https://github.com/tensorflow/tensorflow/files/418658/error.txt)

**Comment by StructML** [comment]
Hello sherrym,

I managed to get it to read the checkpoint file, but now its outputting:

W tensorflow/core/util/tensor_slice_reader.cc:95] Could not open /Users/Zanhuang/Desktop/NNP/checkpoint: Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?

over and over again for the number of variables in that file.

What seems to be the problem here?

**Comment by sherrym** [comment]
Hi @StructML , I don't think your code is able to read the correct checkpoint file. Could you paste the exact commands you used for training and eval here? Also, please paste the output of the following command here:

# find /Users/Zanhuang/Desktop/NNP/

**Comment by StructML** [comment]
For Training:

`import Input
import Process

import time
import numpy as np
import os

import tensorflow as tf
from datetime import datetime

FLAGS = tf.app.flags.FLAGS

def train():
    with tf.Session() as sess:
        images, labels = Process.inputs()

        forward_propgation_results = Process.forward_propagation(images)

        train_loss, cost = Process.error(forward_propgation_results, labels)

        image_summary_t = tf.image_summary(images.name, images, max_images = 2)

        summary_op = tf.merge_all_summaries()

        init = tf.initialize_all_variables()

        saver = tf.train.Saver()

        sess.run(init)

        saver = tf.train.Saver(tf.all_variables())

        tf.train.start_queue_runners(sess = sess)

        train_dir = "/Users/Zanhuang/Desktop/NNP/model.ckpt"

        summary_writer = tf.train.SummaryWriter(train_dir, sess.graph)

        for step in range(100):
            start_time = time.time()
            print(sess.run([train_loss, cost]))
            duration = time.time() - start_time
            if step % 1 == 0:
                num_examples_per_step = FLAGS.batch_size
                examples_per_sec = num_examples_per_step / duration
                sec_per_batch = float(duration)

                format_str = ('%s: step %d, (%.1f examples/sec; %.3f ''sec/batch)')
                print (format_str % (datetime.now(), step, examples_per_sec, sec_per_batch))

                summary_str = sess.run(summary_op)
                summary_writer.add_summary(summary_str, step)

                if step % 2 == 0:
                    checkpoint_path = os.path.join(FLAGS.data_dir, 'model.ckpt')
                    saver.save(sess, checkpoint_path, global_step = step)


def main(argv = None):
    train()

if __name__ == '__main__':
  tf.app.run()`

For Eval:

`import main
import Process
import Input

eval_dir = "/Users/Zanhuang/Desktop/NNP/checkpoint"
checkpoint_dir = "/Users/Zanhuang/Desktop/NNP"


def evaluate():
  with tf.Graph().as_default() as g:
    images, labels = Process.eval_inputs()
    forward_propgation_results = Process.forward_propagation(images)
    init_op = tf.initialize_all_variables()
    saver = tf.train.Saver()
    top_k_op = tf.nn.in_top_k(forward_propgation_results, labels, 1)

  with tf.Session(graph = g) as sess:
    tf.train.start_queue_runners(sess = sess)
    sess.run(init_op)
    saver.restore(sess, eval_dir)
    for i in range(100):
        print(sess.run(top_k_op))

def main(argv = None):
    evaluate()

if __name__ == '__main__':
  tf.app.run()
`

The results of find is:

/Users/Zanhuang/Desktop/NNP/
/Users/Zanhuang/Desktop/NNP//.DS_Store
/Users/Zanhuang/Desktop/NNP//checkpoint
/Users/Zanhuang/Desktop/NNP//DATA_PREPARE
/Users/Zanhuang/Desktop/NNP//TRAIN_MODI/TEST
/Users/Zanhuang/Desktop/NNP//TRAIN_MODI/TEST/.DS_Store
/Users/Zanhuang/Desktop/NNP//TRAIN_MODI/TEST/enput.py
/Users/Zanhuang/Desktop/NNP//TRAIN_MODI/TEST/Prostate_Cacer_Data1.bin
/Users/Zanhuang/Desktop/NNP//TRAIN_MODI/TEST/Prostate_Cancer_Data1.bin
/Users/Zanhuang/Desktop/NNP//TRAIN_MODI/TEST/Prostate_Cancer_Data2.bin
/Users/Zanhuang/Desktop/NNP//TRAIN_MODI/TEST/Prostate_Cancer_Data3.bin
/Users/Zanhuang/Desktop/NNP//TRAIN_MODI/TEST/Prostate_Cancer_Data4.bin

**Comment by sherrym** [comment]
In your training code, you had specified "/Users/Zanhuang/Desktop/NNP/model.ckpt" to be the directory for writing checkpoints. I do not see that directory under /Users/Zanhuang/Desktop/NNP at all.

Please read my previous comment on how to use train_dir, eval_dir and checkpoint_dir.
============================================================
Hi there. I think the confusion came from how train_dir, eval_dir and checkpoint_dir are used. You need to set your checkpoint_dir to where your checkpoints reside. All the *_dir need to be directories instead of files.

Please see

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/models/image/cifar10/cifar10_train.py
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/models/image/cifar10/cifar10_eval.py

for examples.

============================================================

**Comment by StructML** [comment]
Strange, there is however a model.ckpt folder.

**Comment by StructML** [comment]
It still doesn't work even after moving the checkpoints out of that directory in to NNP.

**Comment by prb12** [comment]
This looks to me like all of your paths are confused:

In your new training code, you have:
```        train_dir = "/Users/Zanhuang/Desktop/NNP/model.ckpt" ```
You're writing checkpoints to here:
 ```       checkpoint_path = os.path.join(FLAGS.data_dir, 'model.ckpt') ```

Where `FLAGS.data_dir` is unspecified... do you provide this on the command line?  If not, what is the default value? 

When doing eval, you have:
```       eval_dir = "/Users/Zanhuang/Desktop/NNP/checkpoint" ```
but strangely
```       checkpoint_dir = "/Users/Zanhuang/Desktop/NNP"  ```

And you're actually restoring from: 
```       saver.restore(sess, eval_dir) ```

There does not appear to be a bug here, so I'm going to close this issue.  
If you are still having problems making your code work, StackOverflow is a more appropriate venue for general usage questions.
